libbeat/cmd/instance/metrics.go

```diff
@@ -1,16 +1,39 @@
 package instance
 
 import (
+	"os"
 	"runtime"
 	"time"
 
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/monitoring"
+	sigar "github.com/elastic/gosigar"
+)
+
+type cpuSample struct {
+	time      time.Time
+	procTimes sigar.ProcTime
+}
+
+var (
+	numCores   = runtime.NumCPU()
+	lastSample = cpuSample{
+		time:      time.Now(),
+		procTimes: sigar.ProcTime{},
+	}
 )
 
 func init() {
+	pid := os.Getpid()
+	err := lastSample.procTimes.Get(pid)
+	if err != nil {
+		panic(err)
```

Comment:
I think you should handle the error rather than `panic`'ing. I would reserve the use of panics to cases that are completely unrecoverable or are programmer errors that are covered by unit tests (like hitting the `default` case in a switch statement that is supposed to cover all enum values). Panics on init can be really hard to debug especially on Windows.